engine/classes/Elgg/AutoloadManager.php

```diff
@@ -193,5 +193,9 @@ public function getLoader() {
 	public function setStorage(\ElggCache $storage) {
 		$this->storage = $storage;
 	}
+
+	public function __destruct() {
```

Comment:
This may work, but I've had several bad experiences relying on PHP destructors, particularly calling out to other objects (that may have already been destroyed).